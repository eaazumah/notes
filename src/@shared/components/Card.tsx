import React from 'react';
import styled from 'styled-components/native';
import Parser from '../../parser/Parser';
import {deviceWidth} from '../constants';
import capitalize from '../helpers/capitalize';

interface ICardItem {
  id: string;
  createdAt: number;
  title: string;
  text: string;
  favorite: boolean;
}

interface ICardProps {
  item: ICardItem;
}

const Card: React.FC<ICardProps> = props => {
  const title = capitalize(props.item.title);
  return (
    <Container activeOpacity={0.8}>
      <Title numberOfLines={1}>{title}</Title>
      <Parser numberOfLines={10} opacity={0.8}>
        {props.item.text}
      </Parser>
    </Container>
  );
};

const Container = styled.TouchableOpacity`
  padding: 10px;
  border-radius: 10px;
  margin-bottom: 20px;
  margin-horizontal: 7px;
  width: ${deviceWidth / 2.3}px;
  border-width: 1px;
  border-color: ${props => props.theme.surface};
`;

const Title = styled.Text`
  font-size: 15px;
  margin-top: 5px;
  margin-bottom: 10px;
  font-weight: bold;
  color: ${props => props.theme.primaryText};
`;

// const CreatedAt = styled.Text`
//   font-size: 12px;
//   color: ${props => props.theme.primary};
// `;

export default Card;
