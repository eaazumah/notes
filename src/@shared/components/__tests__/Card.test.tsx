import {cleanup, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../../__testUtils__/Providers';
import Card from '../Card';

describe('Card', () => {
  afterEach(cleanup);

  test('Card renders correctly', async () => {
    const item = {
      id: 'kd35y6',
      title: 'Title',
      text: 'My notes',
      favorite: false,
      createdAt: Date.now(),
    };
    const onPress = jest.fn();
    const container = render(
      <TestProvider>
        <Card item={item} onPress={onPress} />
      </TestProvider>,
    );
    expect(container.toJSON()).toMatchSnapshot();
  });

  test('Card functionality test', async () => {
    const item = {
      id: 'kd35y6',
      title: 'Title',
      text: 'My notes',
      favorite: false,
      createdAt: Date.now(),
    };
    const onCardPress = jest.fn();

    const container = render(
      <TestProvider>
        <Card item={item} onPress={onCardPress} />
      </TestProvider>,
    );

    const component = await container.findByTestId('card');
    expect(component).toBeTruthy();
    expect(onCardPress).not.toHaveBeenCalled();
    fireEvent.press(component);
    expect(onCardPress).toHaveBeenCalled();
  });
});
