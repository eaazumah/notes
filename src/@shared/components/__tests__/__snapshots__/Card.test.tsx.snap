// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#e6e6e6",
      "borderRadius": 10,
      "borderWidth": 1.5,
      "marginBottom": 20,
      "marginHorizontal": 7,
      "maxHeight": 200,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
  testID="card"
>
  <HTMLView
    maxHeight={200}
    rootComponentProps={
      Object {
        "maxHeight": 200,
        "overflow": "hidden",
      }
    }
    style={
      Array [
        Object {},
      ]
    }
    textComponentProps={
      Object {
        "style": Object {
          "color": "#242A37",
        },
      }
    }
    value="My notes"
  />
</View>
`;
